### PR TITLE
fix: config options like `--port` or `--config` are ignored

### PR DESCRIPTION
**Pull request by bugra9**
### New Pull Request Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/parse-dashboard/issues?q=is%3Aissue).

Related issue: #2112

### Approach

> https://github.com/tj/commander.js/releases/tag/v7.0.0
> **Migration Tips**
> **program options**
> Use the .opts() method to access the options. This is available on any command but is used most with the program.

### TODOs before merging

- [x] Add tests
- [ ] Add changes to documentation (guides, repository pages, in-code descriptions)
- [x] A changelog entry is created automatically using the pull request title (do not manually add a changelog entry)

### CODE REVIEW

**Review by mtrezza — DISMISSED** [review]

**Comment by mtrezza** [comment]
@bugra9 Is this ready for merge or do you want to add anything? Do you think it would be possible to also add a test to ensure this issue is detected in our CI in the future?

**Comment by bugra9** [comment]
@mtrezza I added a test to ensure port, config, appId, serverURL, masterKey, appName, graphQLServerURL options are not ignored.

`yarn lint` and `yarn test` commands ran successfully on my local system. If CI tests passes successfully, it is ready to merge for me.

#2066 needs to be updated after this merge request has been merged.
`const allowAnonymousUser = program.allowAnonymousUser || process.env.PARSE_DASHBOARD_ALLOW_ANONYMOUS_USER`

**Comment by mtrezza** [comment]
The repo uses npm. Could you please make sure the same commands run fine with npm?

I also see that you are still making commits, so kindly re-request a review from the `dashboard-review` team once you think this is done.

**Comment by bugra9** [comment]
I couldn't start CI so when I requested a review I didn't know that CI wouldn't work.
> First-time contributors need a maintainer to approve running workflows.
 
My mistake, it didn't occur to me to run the tests on the fork repo.

Now, I think this is done.
- `npm run lint` and `npm run test` commands ran successfully on my local system (node 12, 14 and 16). 
- All checks have passed on fork repo.

**Comment by mtrezza** [comment]
Yes, the CI will run automatically on your future PRs. I just started it, let's see whether all passes.

File: package.json
Comment (mtrezza) [inline package.json]: The tests are actually under `/src/lib/tests`. Could you move the test to there?

I suspect that you can actually move the test into one of the existing files instead of creating a new file, if it fits thematically:
- `src/lib/tests/e2e/dashboard.e2e.test.js` or 
- `src/lib/tests/e2e.js`

**Review by mtrezza — CHANGES_REQUESTED** [review]

File: Parse-Dashboard/tests/.eslintrc.json
Comment (mtrezza) [inline Parse-Dashboard/tests/.eslintrc.json]: Not needed, already exists under `src/lib/tests/.eslintrc.json`

File: package.json
Comment (bugra9) [reply package.json]: I moved the test to src/lib/tests/e2e/dashboard.e2e.test.js.

File: Parse-Dashboard/tests/.eslintrc.json
Comment (bugra9) [reply Parse-Dashboard/tests/.eslintrc.json]: I deleted it.

File: src/lib/tests/e2e/dashboard.e2e.test.js
Comment (mtrezza) [inline src/lib/tests/e2e/dashboard.e2e.test.js]: Please change to:
```
describe('Config options', (
```

File: src/lib/tests/e2e/dashboard.e2e.test.js
Comment (mtrezza) [inline src/lib/tests/e2e/dashboard.e2e.test.js]: Please change to:

```
it('should start with port option', async () => {
```

File: src/lib/tests/e2e/dashboard.e2e.test.js
Comment (mtrezza) [inline src/lib/tests/e2e/dashboard.e2e.test.js]: Could you combine all the new tests that should yield the same error message into a single test?

```js
it('should reject to start if config and other options are combined', async () => {
...

const args = [
  '--appId',
  '--serverURL',
  ...
];

for (const arg of args) {
  const result = await startParseDashboardAndGetOutput(['--config', 'helloworld', arg, 'helloworld']);
  expect(...)
}
```

File: src/lib/tests/e2e/dashboard.e2e.test.js
Comment (mtrezza) [inline src/lib/tests/e2e/dashboard.e2e.test.js]: Rename to `timeoutInMs` and remove comment, we use camelcase for vars.

**Review by mtrezza — CHANGES_REQUESTED** [review]

File: src/lib/tests/e2e/dashboard.e2e.test.js
Comment (bugra9) [reply src/lib/tests/e2e/dashboard.e2e.test.js]: I changed it.

File: src/lib/tests/e2e/dashboard.e2e.test.js
Comment (bugra9) [reply src/lib/tests/e2e/dashboard.e2e.test.js]: I changed it.

**Review by mtrezza — APPROVED** [review]
Looks good! Let's wait for CI to pass.

File: src/lib/tests/e2e/dashboard.e2e.test.js
Comment (bugra9) [reply src/lib/tests/e2e/dashboard.e2e.test.js]: Done. It looks nice now.

File: src/lib/tests/e2e/dashboard.e2e.test.js
Comment (bugra9) [reply src/lib/tests/e2e/dashboard.e2e.test.js]: Done

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [4.1.2-alpha.2](https://github.com/parse-community/parse-dashboard/releases/tag/4.1.2-alpha.2)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [4.1.2-beta.1](https://github.com/parse-community/parse-dashboard/releases/tag/4.1.2-beta.1)

**Comment by fdddf** [comment]
This issue still exist in version parse-dashboard 5.0.0
The only way to work is give a particular port with --port:
```
parse-dashboard --config dashboard.json  --port 8080
```

**Comment by mtrezza** [comment]
Could you please open a new issue if the issue still exists? This PR is closed.

**Comment by fdddf** [comment]
> Could you please open a new issue if the issue still exists? This PR is closed.

Okay, I will open another issue.

**Pull request merged**
